03/app.js

```diff
@@ -10,18 +10,43 @@ class App extends React.Component {
     state = {
         cart: [],
     }
-    
+
     render() {
         return (
             <section>
-                <Category />
-                <Cart />
+                <Category data={data} cart={this.state.cart} onClick={this.addToCart} />
+                <Cart cart={this.state.cart} onClick={this.removeFromCart} />
             </section>
         )
     }
+
+    addToCart = (product) => {
+        this.setState( state => {
+            return {
+                cart: [...state.cart, product]
```

Comment:
Zamiast zapisywać wszystkie dane i je powielać to wystarczy zapisać tylko `id`, ale wtedy zmienia się trochę logika. Nie sugeruję, aby to teraz zmieniać, ale pamiętać na przyszłość ;)